sql/auth/sql_auth_cache.cc

```diff
@@ -1606,11 +1606,13 @@ acl_init_utility_user(bool check_no_resolve)
     {
       if (*cur_pos == ',' || *cur_pos == '\0')
       {
-        acl_utility_user_schema_access.push_back(std::string(cur_db,
-                                                             cur_pos
-                                                             - cur_db));
+	if (cur_pos - cur_db > 1) {
```

Comment:
Shouldn't there be `cur_pos - cur_db >= 1`?

Reply:
No, that still would include the empty entries - if the length of this string is one, it is the comma, or the end of string character.

Reply:
consider the input `a,b`
when `cur_pos` points to the comma `,`,  `cur_db` points to the beginning of the string `a`
in this case `cur_pos - cur_db == 1`, we won't gen into this branch

Reply:
Sorry, you are right. And I did this correctly in 5.6, it's only incorrect on 5.7...